numba/tests/test_npdatetime.py

```diff
@@ -541,7 +541,12 @@ def check(a):
     def test_hash(self):
         f = self.jit(hash_usecase)
         def check(a):
-            self.assertPreciseEqual(f(a), hash(a))
+            # Generic timedeltas (those without a unit)
+            # are no longer hashable beyond NumPy 2.2
+            # Non-generic timedeltas will have dtype name
+            # as timedelta64[<unit>]
+            if numpy_version < (2, 2) and a.dtype.name != 'timedelta64':
+                self.assertPreciseEqual(f(a), hash(a))
```

Comment:
```suggestion
            if not (numpy_version >= (2, 2) and a.dtype.name == 'timedelta64'):
                self.assertPreciseEqual(f(a), hash(a))
```
Is this testing the right combination? This should continue to check all existing NumPy versions `hash` `timedelta64` regardless of whether they are generic and only skip the combination of 2.2 + generic?